observatory-platform/observatory/platform/utils/gc_utils.py

```diff
@@ -1232,3 +1234,136 @@ def delete_bucket_dir(*, bucket_name: str, prefix: str):
+    bucket = storage_client.get_bucket(bucket_name)
+
+    # Timezone of the bucket is in UTC. Have to change local timezone to UTC to get correct age.
+    days_old = (datetime.datetime.now(datetime.timezone.utc) - bucket.time_created).days
+
+    return days_old
+
+
+def get_age_of_dataset_in_days(project_id: str, dataset_id: str):
+
+    """Determines the age of a dataset in days from time of created to present
+    :param project_id: project_name
+    :param bucket_name: Name of bucket
+    :result age_in_days: Age of the bucket in days"""
+
+    client = bigquery.Client(project=project_id)
+    dataset = client.get_dataset(dataset_id)
+
+    # Timezone of the dataset is in UTC. Have to change local timezone to UTC to get correct age.
+    days_old = (datetime.datetime.now(datetime.timezone.utc) - dataset.created).days
+
+    return days_old
+
+def delete_old_buckets_with_prefix(project_id: str, prefix: str, age_to_delete: int = 7):
+    """Deletes buckets that share the same prefix and if it is older than "age_to_delete" days.
+
+    :param project_id: project_id that you wish to delete buckets from.
+    :param prefix: The identifying prefix of the prject and telescope name.
+    :param age_to_delete: Delete if the age of the dataset is older than this amount.
+    """
+
+    client = storage.Client(project=project_id)
+
+    # List all buckets in the project.
+    bucket_list = list_all_buckets()
+
+    buckets_deleted = []
+    for bucket_id in bucket_list:
+
+        # Check if prefix is in the bucket name.
+        if (prefix in bucket_id) and (prefix != ""):
+
+            # Check bucket age
+            bucket_age = get_age_of_bucket_in_days(project_id, bucket_id)
+
+            # Delete bucket if older than 7 days.
+            if bucket_age >= age_to_delete:
+                
+                bucket = client.get_bucket(bucket_id)
+                bucket.delete(force=True)
+
+                buckets_deleted.append(bucket_id)
+
+    if len(buckets_deleted) < 1:
+        logging.info(f"No buckets older than {age_to_delete} days to delete.")
+    else:
+        logging.info(f"Deleted the following buckets older than {age_to_delete} days: {buckets_deleted}")
+
+    return buckets_deleted
+
+def delete_old_datasets_with_prefix(project_id: str, prefix: str, age_to_delete: int = 7):
+    """Deletes datasets that share the same prefix and if it is older than "age_to_delete" days.
+
+    :param project_id: project_id that you wish to delete datasets from.
+    :param dataset_id: dataset_id that you wish to remove.
+    :param age_to_delete: Delete if the age of the dataset is older than this amount.
+    """
+
+    client = bigquery.Client(project=project_id)
+
+    # List all datsets in the project.
+    dataset_list = list_all_datasets()
+
+    datasets_deleted = []
+    for dataset_id in dataset_list:
+
+        # Check if prefix is in the dataset name.
+        if (prefix in dataset_id) and (prefix != ""):
```

Comment:
It would be safer to call `dataset_id.startswith(prefix)` so that only datasets that start with the prefix are deleted, not datasets with a prefix anywhere in the string.